chalicelib/action.py

```diff
@@ -357,14 +357,31 @@ def perform_action(self):
                 message=f"Delete doesn't support date range :cry:"
             )
 
-        if not self._check_locks(date=date["from"], second_date=date["to"]):
-            self.send_attachment(
-                attachment=delete_message_menu(self.user_name, date_string)
+        if self._check_locks(date=date["from"], second_date=date["to"]):
+            return self.send_response(
+                message=f"Unable to delete since month is locked :cry:"
             )
-        else:
-            self.send_response(message=f"Unable to delete since month is locked :cry:")
+
+        if not self._has_events(date["from"]):
```

Comment:
We have `_get_events()` method that we might use instead.

And then you can remove the `_has_events` method, or simplify it by reusing `_get_events()` in that method